packages/@aws-cdk/aws-apigateway/lib/domain-name.ts

```diff
@@ -162,3 +170,23 @@ export interface DomainNameAttributes {
    */
   readonly domainNameAliasHostedZoneId: string;
 }
+
+/**
+ * The mutual TLS authentication configuration for a custom domain name.
+ */
+export interface MutualTlsAuthenticationAttributes {
+  /**
+   * An Amazon S3 URL that specifies the truststore for mutual TLS authentication, for example, s3://bucket-name/key-name.
+   * The truststore can contain certificates from public or private certificate authorities.
+   * To update the truststore, upload a new version to S3, and then update your custom domain name to use the new version.
+   * To update the truststore, you must have permissions to access the S3 object.
+   */
+  readonly truststoreUri: string;
+
+  /**
+   *  The version of the S3 object that contains your truststore.
+   *  To specify a version, you must have versioning enabled for the S3 bucket.
+   *  @default - Uses current version
+   */
+  readonly truststoreVersion?: string;
+}
```

Comment:
Since this is a reference to an S3 object, this can be - 

```ts
export interface MTLSAttributes {
  readonly bucket: s3.IBucket;
  readonly key: string;
  readonly version?: string;
}
```

Reply:
I was just starting to refactor towards this as well. Good to know I was going down the right path.